src/pages/Governance/ProposalList/CreateProposalModal/index.tsx

```diff
@@ -40,86 +44,162 @@ export const CreateProposal: React.FC<CreateProposalProps> = ({
+            css={styles.formBottomMargin}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
 
-            <FormikTextField
-              name="abstainDescription"
-              placeholder={t('vote.createProposalForm.abstainOption')}
-              label={t('vote.abstain')}
-              css={styles.sectionSpacing}
-              displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
-            />
-          </>
-        ),
-      },
-      {
-        title: t('vote.pages.actions'),
-        Component: () => <ActionAccordion />,
-      },
+          <FormikMarkdownEditor
+            name="description"
+            placeholder={t('vote.createProposalForm.addDescription')}
+            label={t('vote.createProposalForm.description')}
+            css={styles.sectionSpacing}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+        </>
+      ),
+    },
+    {
+      title: t('vote.pages.votingOptions'),
+      Component: () => (
+        <>
+          <FormikTextField
+            name="forDescription"
+            placeholder={t('vote.createProposalForm.forOption')}
+            label={t('vote.for')}
+            css={styles.formBottomMargin}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+
+          <FormikTextField
+            name="againstDescription"
+            placeholder={t('vote.createProposalForm.againstOption')}
+            label={t('vote.against')}
+            css={styles.formBottomMargin}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+
+          <FormikTextField
+            name="abstainDescription"
+            placeholder={t('vote.createProposalForm.abstainOption')}
+            label={t('vote.abstain')}
+            css={styles.sectionSpacing}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+        </>
+      ),
+    },
+    {
+      title: t('vote.pages.actions'),
+      Component: () => <ActionAccordion />,
+    },
+    {
+      title: t('vote.pages.confirmation'),
+      Component: () => <ProposalPreview />,
+    },
+  ];
+
+  const handleImportProposal = async (file: File | null, formikContext: ProposalFormContext) => {
+    const { setFieldValue, validateForm } = formikContext;
+    if (file) {
+      try {
+        const {
+          actions,
+          proposalType,
+          title,
+          description,
+          forDescription,
+          abstainDescription,
+          againstDescription,
+        } = await importJsonProposal(file);
+
+        setFieldValue('actions', actions);
+        setFieldValue('title', title);
+        setFieldValue('proposalType', proposalType);
+        setFieldValue('description', description);
+        setFieldValue('forDescription', forDescription);
+        setFieldValue('abstainDescription', abstainDescription);
+        setFieldValue('againstDescription', againstDescription);
+        await validateForm();
```

Comment:
```suggestion
        const proposal = await importJsonProposal(file);
        setFieldValues(proposal);
```

`setFieldValues` will automatically trigger a revalidation of the form.